src/pages/index.js

```diff
@@ -72,19 +72,24 @@ function handleSubmit(
   request,
   popupInstance,
   loadingText = "Saving...",
-  defaultText = "Save"
+  defaultText = "Save",
+  successCallback
 ) {
   popupInstance.setLoading(true, loadingText);
 
   return request()
     .then(() => {
       popupInstance.close();
+      popupInstance.reset();
     })
     .catch((error) => {
       console.error("Error:", error);
     })
     .finally(() => {
       popupInstance.setLoading(false, defaultText);
+      if (successCallback) {
+        successCallback();
```

Comment:
` successCallback` can be called only after `successful` response from the server in block `then`, so you don't need it here. It can be in `then` which is in `makeRequest` before you pass it into `handleSubmit`